### PR TITLE
DEVX-1966 DEVX-1968 DEVX-1972: Docs conversation: groovy spring kotlin

### PR DESCRIPTION
**Pull request by tikimims**
Migrated the following languages from AsciiDoc to RST.

- Java Spring Boot
- Kotlin
- Groovy

### CODE REVIEW

File: clients/docs/clients-all-examples.rst [hunk 1]
```diff
@@ -40,5 +40,8 @@ For the subset of languages that support it, the code examples also demonstrate
       ccloud
       csharp
       go
+      groovy
+      java-springboot
```
Comment (ybyzek) [inline clients/docs/clients-all-examples.rst]: Springboot should follow java

File: clients/docs/kotlin.rst [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+#. .. include:: includes/client-example-create-file-java.rst
+
+
+Basic Producer and Consumer
+---------------------------
+
+.. include:: includes/producer-consumer-description.rst
+
+Produce Records
+~~~~~~~~~~~~~~~~
```
Comment (ybyzek) [inline clients/docs/kotlin.rst]:
```suggestion
~~~~~~~~~~~~~~~
```

File: clients/docs/java-springboot.rst [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+         2020-02-13 14:41:58.248  INFO 44191 --- [ntainer#0-0-C-1] i.c.e.c.c.springboot.ConsumerExample     : received alice {"count": 2}
+         2020-02-13 14:41:58.248  INFO 44191 --- [ntainer#0-0-C-1] i.c.e.c.c.springboot.ConsumerExample     : received alice {"count": 3}
+         2020-02-13 14:41:58.249  INFO 44191 --- [ntainer#0-0-C-1] i.c.e.c.c.springboot.ConsumerExample     : received alice {"count": 4}
+         2020-02-13 14:41:58.249  INFO 44191 --- [ntainer#0-0-C-1] i.c.e.c.c.springboot.ConsumerExample     : received alice {"count": 5}
+         2020-02-13 14:41:58.249  INFO 44191 --- [ntainer#0-0-C-1] i.c.e.c.c.springboot.ConsumerExample     : received alice {"count": 6}
+         2020-02-13 14:41:58.249  INFO 44191 --- [ntainer#0-0-C-1] i.c.e.c.c.springboot.ConsumerExample     : received alice {"count": 7}
+         2020-02-13 14:41:58.249  INFO 44191 --- [ntainer#0-0-C-1] i.c.e.c.c.springboot.ConsumerExample     : received alice {"count": 8}
+         2020-02-13 14:41:58.249  INFO 44191 --- [ntainer#0-0-C-1] i.c.e.c.c.springboot.ConsumerExample     : received alice {"count": 9}
+
+#. When you are done, press ``kbd:[CTRL-C]``.
```
Comment (ybyzek) [inline clients/docs/java-springboot.rst]: Should this read just ``<ctrl-c>``?

File: clients/docs/java-springboot.rst [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+#. View the :devx-examples:`producer|clients/cloud/java-springboot/src/main/java-springboot/io/confluent/examples/clients/cloud/springboot/kafka/ProducerExample.java` and :devx-examples:`consumer code|clients/cloud/java-springboot/src/main/java-springboot/io/confluent/examples/clients/cloud/springboot/kafka/ConsumerExample.java`.
```
Comment (ybyzek) [inline clients/docs/java-springboot.rst]:
```suggestion
#. View the :devx-examples:`producer code|clients/cloud/java-springboot/src/main/java-springboot/io/confluent/examples/clients/cloud/springboot/kafka/ProducerExample.java` and :devx-examples:`consumer code|clients/cloud/java-springboot/src/main/java-springboot/io/confluent/examples/clients/cloud/springboot/kafka/ConsumerExample.java`.
```

File: clients/docs/groovy.rst [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+      ./gradlew clean build
+
+#. Run the producer, passing in arguments for:
+
+   - the local file with configuration parameters to connect to your Kafka cluster
+   - the topic name
+
+   .. code-block:: text
+
+         ./gradlew runApp -PmainClass="io.confluent.examples.clients.cloud.ProducerExample" \
```
Comment (ybyzek) [inline clients/docs/groovy.rst]:
```suggestion
       ./gradlew runApp -PmainClass="io.confluent.examples.clients.cloud.ProducerExample" \
```

File: clients/docs/groovy.rst [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+       -PconfigPath="$HOME/.confluent/java.config" \
```
Comment (ybyzek) [inline clients/docs/groovy.rst]:
```suggestion
          -PconfigPath="$HOME/.confluent/java.config" \
```

File: clients/docs/groovy.rst [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+       -Ptopic="test1"
```
Comment (ybyzek) [inline clients/docs/groovy.rst]:
```suggestion
          -Ptopic="test1"
```

File: clients/docs/java-springboot.rst [hunk 8]
```diff
@@ -0,0 +1,176 @@
+.. _client-examples-java-springboot:
+
+Java Spring Boot
```
Comment (ybyzek) [inline clients/docs/java-springboot.rst:3]: The java springboot page isn't rendering quite right.  Let's enable you to build locally so you can generate the docs on your laptop tomorrow.

File: clients/docs/java-springboot.rst [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+
+#. .. include:: includes/clients-checkout.rst
+
+#. Change directory to the example for Java Spring Boot.
+
+   .. code-block:: bash
+
+      cd clients/cloud/java-springboot/
+
+#. .. include:: includes/client-example-create-file-java.rst
```
Comment (ybyzek) [inline clients/docs/java-springboot.rst]: We need a new include for springboot ;)  Let's call it `includes/client-example-create-file-springboot.rst`?

File: clients/docs/java-springboot.rst [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+
+.. admonition:: Confluent Cloud config example
```
Comment (ybyzek) [inline clients/docs/java-springboot.rst]: Lines 39-50 can be deleted

File: clients/docs/java-springboot.rst [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+      ssl.endpoint.identification.algorithm=https security.protocol=SASL_SSL
+      sasl.mechanism=PLAIN
+      sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule
+      required username\=“<API KEY>” password\=“<API SECRET>”;
+      schema.registry.url=<SR ENDPOINT>
+      basic.auth.credentials.source=USER_INFO
+      schema.registry.basic.auth.user.info=<SR_KEY:SR_PASSWORD></programlisting>
+
+
+Avro Producer and Consumer
```
Comment (ybyzek) [inline clients/docs/java-springboot.rst]:
```suggestion
Avro and Confluent Cloud Schema Registry
```

File: clients/docs/java-springboot.rst [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+---------------------------
+
+.. include:: includes/client-example-schema-registry-3.rst
```
Comment (ybyzek) [inline clients/docs/java-springboot.rst]:
```suggestion
.. include:: includes/producer-consumer-description.rst
```

File: clients/docs/java-springboot.rst [hunk 13]
```diff
@@ -0,0 +1,176 @@
...
+
+#. .. include:: includes/client-example-schema-registry-1.rst
+
+#. .. include:: includes/client-example-schema-registry-2.rst
+
+
+This Spring Boot application has the following two components:
+
+  - Producer (``ProducerExample.java``)
```
Comment (ybyzek) [inline clients/docs/java-springboot.rst]:
```suggestion
- :devx-examples:`Producer|clients/cloud/java-springboot/src/main/java/io/confluent/examples/clients/cloud/springboot/kafka/ProducerExample.java`
```

File: clients/docs/java-springboot.rst [hunk 14]
```diff
@@ -0,0 +1,176 @@
...
+  - Consumer (``ConsumerExample.java``)
```
Comment (ybyzek) [inline clients/docs/java-springboot.rst]:
```suggestion
- :devx-examples:`Consumer|clients/cloud/java-springboot/src/main/java/io/confluent/examples/clients/cloud/springboot/kafka/ConsumerExample.java`
```

File: clients/docs/java-springboot.rst [hunk 15]
```diff
@@ -0,0 +1,176 @@
...
+         [Running count]: alice, 6
+         [Running count]: alice, 10
+         [Running count]: alice, 15
+         [Running count]: alice, 21
+         [Running count]: alice, 28
+         [Running count]: alice, 36
+         [Running count]: alice, 45
+         ...
+
+#. When you are done, press ``kbd:[CTRL-C]``.
```
Comment (ybyzek) [inline clients/docs/java-springboot.rst]: Should this read just `<ctrl-c>`?

**Review by ybyzek — CHANGES_REQUESTED** [review]

**Comment by ybyzek** [comment]
@tikimims please rebase and pull in changes from https://github.com/confluentinc/examples/pull/743 .  Note that some of the include file names have changed.

File: clients/docs/clients-all-examples.rst (hunk 1 above)
Comment (tikimims) [reply clients/docs/clients-all-examples.rst]: Updated

File: clients/docs/kotlin.rst (hunk 2 above)
Comment (tikimims) [reply clients/docs/kotlin.rst]: updated

File: clients/docs/java-springboot.rst (hunk 15 above)
Comment (tikimims) [reply clients/docs/java-springboot.rst]: Correct! Good catch. Syntax error with the migration

File: clients/docs/java-springboot.rst (hunk 4 above)
Comment (tikimims) [reply clients/docs/java-springboot.rst]: Updated

File: clients/docs/java-springboot.rst (hunk 3 above)
Comment (tikimims) [reply clients/docs/java-springboot.rst]: Yup good catch

File: clients/docs/java-springboot.rst (hunk 13 above)
Comment (tikimims) [reply clients/docs/java-springboot.rst]: Updated

File: clients/docs/java-springboot.rst (hunk 14 above)
Comment (tikimims) [reply clients/docs/java-springboot.rst]: Updated

File: clients/docs/java-springboot.rst (hunk 12 above)
Comment (tikimims) [reply clients/docs/java-springboot.rst]: Updated

File: clients/docs/java-springboot.rst (hunk 10 above)
Comment (tikimims) [reply clients/docs/java-springboot.rst]: deleted

**Review by ybyzek — APPROVED** [review]
@tikimims LGTM, thanks!

**Pull request merged**